pkg/oci/oci.go

```diff
@@ -478,16 +490,85 @@ func newAuthClient(repository string, authOptions *AuthOptions) (*oras_auth.Clie
 	}, nil
 }
 
+func prepareCheckOpts(ctx context.Context, publicKey string) (*cosign.CheckOpts, error) {
+	pubKey, err := signature.LoadPublicKeyRaw([]byte(publicKey), crypto.SHA256)
+	if err != nil {
+		return nil, fmt.Errorf("loading public key: %w", err)
+	}
+
+	rekorPubKeys, err := cosign.GetRekorPubs(ctx)
```

Comment:
Maybe add a comment on what `RekorPubs` are.

Reply:
Mmmm afaiu, rekor and ct logs are only used for keyless signing, which is not our case. So, we can omit them here.